front/src/modules/timesStops/hooks/useOutputTableData.ts

```diff
@@ -65,51 +73,61 @@ function useOutputTableData(
+    });
 
-        if (pathStepKey in pathStepsByUic) {
-          const pathStepId = pathStepsByUic[pathStepKey].id || '';
-          const schedule = scheduleByAt[pathStepId];
-          const scheduleData = computeScheduleData(schedule, selectedTrainSchedule.start_time);
-          const formattedScheduleData = formatScheduleData(scheduleData);
-          const marginsData = nextOpPoint
-            ? computeMargins(
-                simulatedTrain,
-                opPoint,
-                nextOpPoint,
-                selectedTrainSchedule,
-                pathStepId
-              )
-            : null;
-          const calculatedArrival = checkAndFormatCalculatedArrival(scheduleData, opPoint.time);
+    const suggestedOpRows = suggestedOperationalPoints.map((sugOpPoint, sugOpIndex) => {
+      const opPoint = operationPointsByNameCh[`${sugOpPoint.name}-${sugOpPoint.ch}`];
+      if (!opPoint) {
+        return sugOpPoint;
+      }
+      const nextOpPoint = findNextScheduledOpPoint(
+        operationalPoints,
+        pathStepsWithOpPointIndices,
+        sugOpIndex
+      );
+      const pathStepKey = `${sugOpPoint.uic}-${sugOpPoint.ch}`;
 
-          return {
-            ...sugOpPoint,
-            ...formattedScheduleData,
-            onStopSignal: schedule?.on_stop_signal || '',
-            calculatedArrival,
-            calculatedDeparture:
-              opPoint.duration > 0 ? secToHoursString(opPoint.time + opPoint.duration, true) : '',
-            ...marginsData,
-          } as SuggestedOP;
-        }
+      if (pathStepKey in pathStepsByUic) {
+        const pathStepId = pathStepsByUic[pathStepKey].id || '';
+        const schedule = scheduleByAt[pathStepId];
+        const scheduleData = computeScheduleData(schedule, selectedTrainSchedule.start_time);
+        const formattedScheduleData = formatScheduleData(scheduleData);
+        const marginsData = nextOpPoint
+          ? computeMargins(simulatedTrain, opPoint, nextOpPoint, selectedTrainSchedule, pathStepId)
+          : null;
+        const calculatedArrival = checkAndFormatCalculatedArrival(scheduleData, opPoint.time);
+        return {
+          ...sugOpPoint,
+          ...formattedScheduleData,
+          onStopSignal: schedule?.on_stop_signal || '',
+          calculatedArrival,
+          calculatedDeparture:
+            opPoint.duration > 0 ? secToHoursString(opPoint.time + opPoint.duration, true) : '',
+          ...marginsData,
+        } as SuggestedOP;
+      }
 
-        return { ...sugOpPoint, calculatedArrival: secToHoursString(opPoint.time, true) };
-      }),
-    [simulatedTrain, pathProperties, operationalPoints, selectedTrainSchedule, pathLength]
-  );
+      return { ...sugOpPoint, calculatedArrival: secToHoursString(opPoint.time, true) };
+    });
+
+    return suggestedOpRows.map((sugOpPoint) => {
+      const matchingPathStep = pathStepRows.find(
+        (pathStepRow) => sugOpPoint.positionOnPath === pathStepRow.positionOnPath
```

Comment:
I must admit I'm not a fan of using the position on path as a kind of unique identifier… The backend doesn't guarantee that the `positionOnPath` are unique for each OP. Ideally we should also match the trigram/UIC/etc with e.g. `matchPathStepAndOp()` when we're hitting duplicate `positionOnPath` values. But it sounds unlikely to happen with the France infrastructure, so maybe we can just leave this as a TODO or something…

Reply:
Yes, I agree :/ I have a refacto in mind, can we leave it like this for now ?

Reply:
Sure